tests/unit/r2i/preprocessing/normalize.cc

```diff
@@ -0,0 +1,468 @@
+                                    (new float[FRAME_WIDTH * UNSUPPORTED_FRAME_HEIGHT * CHANNELS],
+                                     std::default_delete<float[]>());
+  error = out_frame->Configure (out_data.get(), FRAME_WIDTH,
+                                UNSUPPORTED_FRAME_HEIGHT,
+                                r2i::ImageFormat::Id::RGB);
+
+  error = preprocessing->Apply(in_frame, out_frame);
+  LONGS_EQUAL(r2i::RuntimeError::Code::MODULE_ERROR, error.GetCode());
+}
+
+TEST(Normalize, UnsupportedFormatId) {
+  /* Fill frames with dummy data */
+  unsigned char value = 0;
+  dummy_frame_data = std::shared_ptr<unsigned char>(new unsigned char[FRAME_SIZE],
+                     std::default_delete<const unsigned char[]>());
+
+  unsigned char *in_data = dummy_frame_data.get();
+
+  for (unsigned int i = 0; i < FRAME_SIZE; i++) {
+    in_data[i] = value;
+    value++;
+  }
+
+  error = in_frame->Configure(in_data, FRAME_WIDTH, FRAME_HEIGHT,
+                              r2i::ImageFormat::Id::RGB);
+
+  std::shared_ptr<float> out_data = std::shared_ptr<float>
+                                    (new float[FRAME_WIDTH * FRAME_HEIGHT * CHANNELS],
+                                     std::default_delete<float[]>());
+  error = out_frame->Configure (out_data.get(), FRAME_WIDTH,
+                                FRAME_HEIGHT,
+                                r2i::ImageFormat::Id::BGR);
+  error = preprocessing->Apply(in_frame, out_frame);
+  LONGS_EQUAL(r2i::RuntimeError::Code::MODULE_ERROR, error.GetCode());
+}
+
+TEST(Normalize, NullInputFrame) {
+  in_frame = nullptr;
+  error = preprocessing->Apply(in_frame, out_frame);
+  LONGS_EQUAL(r2i::RuntimeError::Code::NULL_PARAMETER, error.GetCode());
+}
+
+TEST(Normalize, NullOutputFrame) {
+  out_frame = nullptr;
+  error = preprocessing->Apply(in_frame, out_frame);
+  LONGS_EQUAL(r2i::RuntimeError::Code::NULL_PARAMETER, error.GetCode());
+}
+
+TEST(Normalize, NullInputFrameData) {
+  std::shared_ptr<float> out_data = std::shared_ptr<float>
+                                    (new float[FRAME_WIDTH * FRAME_HEIGHT * CHANNELS],
+                                     std::default_delete<float[]>());
+  error = out_frame->Configure (out_data.get(), FRAME_WIDTH,
+                                FRAME_HEIGHT,
+                                r2i::ImageFormat::Id::BGR);
+
+  error = preprocessing->Apply(in_frame, out_frame);
+  LONGS_EQUAL(r2i::RuntimeError::Code::NULL_PARAMETER, error.GetCode());
+}
+
+TEST(Normalize, NullOutputFrameData) {
+  dummy_frame_data = std::shared_ptr<unsigned char>(new unsigned char[FRAME_SIZE],
+                     std::default_delete<const unsigned char[]>());
+  unsigned char *in_data = dummy_frame_data.get();
+  error = in_frame->Configure(in_data, FRAME_WIDTH, FRAME_HEIGHT,
+                              r2i::ImageFormat::Id::RGB);
+
+  error = preprocessing->Apply(in_frame, out_frame);
+  LONGS_EQUAL(r2i::RuntimeError::Code::NULL_PARAMETER, error.GetCode());
+}
```

Comment:
There are other missing tests:
* Invalid format?
* Zero stddev? 
* Output height != input height
* Output width != input width
* Output format != output format
* Input type not uchar
* Output type not float